homeassistant/components/lupusec/config_flow.py

```diff
@@ -0,0 +1,89 @@
+""""Config flow for Lupusec integration."""
+
+import ipaddress
+import logging
+import socket
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Optional(CONF_NAME): str,
+    }
+)
+
+
+class LupusecConfigFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Lupusec config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is not None:
+            host = user_input.get(CONF_HOST)
+            username = user_input.get(CONF_USERNAME)
+            password = user_input.get(CONF_PASSWORD)
+            name = user_input.get(CONF_NAME)
+
+            errors = await validate_configuration(host, username, password, name)
```

Comment:
Currently we are only checking if its a valid host. Can we maybe do a little check to see if we can actually find a device? If we can check if the device is supported?

The config flow is the best way to tell the user they f*cked up. So we should do our due fill to check if this is a valid device to connect to and abort if not.

Reply:
Added connection check.